From 84135f9402569ecbdcdade7f0468e00994d7495d Mon Sep 17 00:00:00 2001
From: Ross Lagerwall <ross.lagerwall@citrix.com>
Date: Mon, 5 Sep 2016 14:24:09 +0100
Subject: [PATCH 2/6] tools/livepatch: Set stdout and stderr unbuffered

Using both stdout and stderr interleaved without newlines can result in
strange output when using line buffered mode (e.g. a terminal) or when
fully buffered (e.g. redirected to a file). Set both to unbuffered mode
to fix this.

Signed-off-by: Ross Lagerwall <ross.lagerwall@citrix.com>
diff --git a/tools/misc/xen-livepatch.c b/tools/misc/xen-livepatch.c
index 7e101dc..c8b05df 100644
--- a/tools/misc/xen-livepatch.c
+++ b/tools/misc/xen-livepatch.c
@@ -328,7 +328,6 @@ int action_func(int argc, char *argv[], unsigned int idx)
         }
 
         printf(".");
-        fflush(stdout);
         usleep(DELAY);
     } while ( ++retry < RETRIES );
 
@@ -414,6 +413,13 @@ int main(int argc, char *argv[])
 {
     int i, j, ret;
 
+    /*
+     * Set stdout and stderr to be unbuffered to avoid having to fflush
+     * when printing without a newline.
+     */
+    setvbuf(stdout, NULL, _IONBF, 0);
+    setvbuf(stderr, NULL, _IONBF, 0);
+
     if ( argc  <= 1 )
     {
         show_help();
